### PR TITLE
Sort plot-schema and add test to track plot-schema changes

### LINKED ISSUE #5588: commit plot-schema like a baseline?
What if we treated the final plot schema like a baseline? You'd have to generate it and commit it in any PR that changes it, otherwise the build would fail, and then the review process would explicitly review the changes to the schema. This would catch situations where we add something to e.g. axes and it impacts axes in indicator, color bars etc etc.

@alexcjohnson thoughts? cam from a chat with @archmoj ... can discuss at the next meeting

### PR DESCRIPTION
**Pull request by archmoj**
Resolves #5588.
In PRs after this when there is a change in attributes (e.g. to implement a new feature), one should simply run 
```sh
npm run schema
```
and commit the `test/plot-schema.json` file.

On CircleCI, that schema file would be tested against the auto-generated baseline at `dist/plot-schema.json`.
In the case of any difference a diff is presented and test should fail. 
We also added a message on CircleCI asking contributors to run `npm run schema` and commit the results in the "test" folder (i.e. similar to baseline changes).

This would help track the changes to the API way before release cycles, right when a feature/documentation PR is submitted or even before.

In addition to above,
1. As a result of this PR the plot-schema object is sorted to be as stable as possible for future comparisons. 
2. The `compress_attributes` browserify transform is now disabled to generate `plotly.js` files in the `build` folder which not only decreases the build & rebuild times but also help make this possible.

@plotly/plotly_js

### CODE REVIEW

File: tasks/util/sort_object.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    var allKeys = Object.keys(obj);
+    allKeys.sort(caseInsensitive);
+
+    var newObj = {};
+    for(var i = 0; i < allKeys.length; i++) {
+        var key = allKeys[i];
+        var v = obj[key];
+        newObj[key] = (typeof v === 'object' && v !== null && !(v instanceof Array)) ?
+            sortObject(v) :
+            newObj[key] = v;
```
Comment (alexcjohnson) [inline tasks/util/sort_object.js:15]:
```suggestion
            v;
```

File: tasks/util/sort_object.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+function sortObject(obj) {
+    var allKeys = Object.keys(obj);
+    allKeys.sort(caseInsensitive);
+
+    var newObj = {};
+    for(var i = 0; i < allKeys.length; i++) {
+        var key = allKeys[i];
+        var v = obj[key];
+        newObj[key] = (typeof v === 'object' && v !== null && !(v instanceof Array)) ?
```
Comment (alexcjohnson) [inline tasks/util/sort_object.js:13]: Actually we need to dig into arrays too, right? We've got stuff in the schema like:

```
      "items": [
       {
        "valType": "number",
        "min": 0,
        "max": 1,
        "editType": "calc"
       },
       {
        "valType": "number",
        "min": 0,
        "max": 1,
        "editType": "calc"
       }
      ],
```

Comment (archmoj) [reply tasks/util/sort_object.js]: Good call.
https://github.com/plotly/plotly.js/pull/5776/commits/aea30228e2fa0a15f8dcbccf728b2553d035470f also fixed cases where arrays e.g. *enumerated* `valType` turned into objects!
Before:
```json
   "values": {
    "0": "forward",
    "1": "reverse"
   }
```
After
```json   
   "values": [
    "forward",
    "reverse"
   ]
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -138,14 +138,19 @@ Three additional helpers exist that are refreshed every second:
...
-which has live-reloading and a bunch of other cool features.
-An online version of `plotly-mock-viewer` is available at <https://rreusser.github.io/plotly-mock-viewer/>
-which uses <https://cdn.plot.ly/plotly-latest.min.js>
+```bash
+npm run schema
+```
+
+#### Step 6: Review & commit potential changes made to test/plot-schema.json
+
+> If you are editing attribute descriptions or implementing a new feature this file located in test folder would record the proposed changes to the API. This test file is different from the other plot-schema.json file located in the dist folder and should only be updated by maintainers at the release time.
```
Comment (alexcjohnson) [inline CONTRIBUTING.md:149]:
```suggestion
> If you are editing attribute descriptions or implementing a new feature this file located in the test folder records the proposed changes to the API. Note that there is another plot-schema.json file located in the dist folder, which should only be updated by the maintainers at release time.
```

**Review by alexcjohnson — APPROVED** [review]
💃

**Pull request merged**